### DISCUSSION TITLE
get internal layer output

### DISCUSSION THREAD

**Discussion by sifourquier** [majianjia/nnom#129] [Q&A]
Hello

I try to get internal layer output.
I have set a callback with model_set_callback it is call at all layer but how to get output of current layer? 

can you help me?

**Comment by majianjia** [comment]
You can modify the `nnom_model_create()`, try to record the layer instance you want to get the output.
Then in the callback, you could compare the instance to the one that passed in.  
The data only valid during the callback and will be overwritten by others after, so you will need to copy it out if you want to process it later.

**Reply by sifourquier** [reply]
Thanks you
I not understant how to found the pointer to the buffer.
But i have found a other solution i have add a layer that concatenad all layer output.
Is realy not optimized for memory usage. But is only for test on computer not in final product.
(and is compatible with tensorflowlite)
